src/Type/Php/ArraySumFunctionDynamicReturnTypeExtension.php

```diff
@@ -0,0 +1,36 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Type\Php;
+
+use PhpParser\Node\Expr\FuncCall;
+use PHPStan\Analyser\Scope;
+use PHPStan\Reflection\FunctionReflection;
+use PHPStan\Type\FloatType;
+use PHPStan\Type\Type;
+use PHPStan\Type\UnionType;
+
+class ArraySumFunctionDynamicReturnTypeExtension implements \PHPStan\Type\DynamicFunctionReturnTypeExtension
+{
+
+	public function isFunctionSupported(FunctionReflection $functionReflection): bool
+	{
+		return $functionReflection->getName() === 'array_sum';
+	}
+
+	public function getTypeFromFunctionCall(FunctionReflection $functionReflection, FuncCall $functionCall, Scope $scope): Type
+	{
+		$argType = $scope->getType($functionCall->args[0]->value);
```

Comment:
Analysing `array_sum()` call will crash on this line.

Reply:
But tests are passing... how come?

Reply:
Because you're not testing calling `array_sum()` without any arguments.